src/path.rs

```diff
@@ -23,3 +30,9 @@ pub fn toolchain_dir() -> PathBuf {
 pub fn toolchain_bin_dir(toolchain: &str) -> PathBuf {
     toolchain_dir().join(toolchain).join("bin")
 }
+
+pub fn ensure_dir_exists(path: &Path) {
+    if !path.is_dir() {
+        fs::create_dir_all(path).unwrap_or_else(|_| panic!("Failed to create {}", path.display()))
```

Comment:
While unlikely, I'm not sure we want to `panic!` here - could we use `or_else` instead and propagate the result?